pkg/deploy/deploy.go

```diff
@@ -0,0 +1,19 @@
+package deploy
+
+import (
+	"context"
+	spinnakerv1alpha2 "github.com/armory/spinnaker-operator/pkg/apis/spinnaker/v1alpha2"
+	"github.com/armory/spinnaker-operator/pkg/generated"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type ManifestGenerator interface {
```

Comment:
This will get injected into all deployers, but the `Generate` method seems too coupled with just deploying Spinnaker (returns a `SpinnakerGeneratedConfig`). What if this interface is just `Halyard` so that all deployers can call any Halyard method if needed? Halyard service instance is also passed to validators using their `Options` object.

Reply:
Yeah that was the intent (make Halyard(ish) service available to all) We could enhance the interface as needed. I'd rather not use `halyard` directly to not couple these two (we have too many of these dependencies already!)